### ISSUE TITLE
WH: } used in formatting shows up ingame

### ISSUE THREAD

**Issue by Kryomancer** [umineko-project/umineko-scripting#15]
Game language: WH (r3617)
Operating system: Windows (onscripter-ru r3567)
Episode and chapter where the issue occurs: Ep 6 Ch 17
https://github.com/umineko-project/umineko-scripting/blob/357a669103036b01a01792bc87a2aaad3da0ad62/story/ep6/wh/umi6_17.txt#L371

Description of the issue:
One of the } used for the markup to center a line displays ingame. This seems to happen only in WH, checked the EN version and this is not present there although the lines seem functionally identical and similar markup elsewhere doesn't display this behaviour:

WH:
```
`{a:c: `
`..."Mother"?} `
```
EN: 
```
`{a:c:`
`......“Mother”?}`
```

Despite this the rest of the text renders normally, not centered. Sorry, I got no clue why this is happening.

Screenshot of the issue
![img](https://i.imgur.com/sbUH4S3.png)

**Comment by vit9696** [comment]
You cannot split it over segments. Need to be added separately per each segment from what I remember. This is the current limitation.

**Comment by ooa113y** [comment]
I have managed to fix this in b96ef19 by blindly copying-and-pasting the lines from the English script into the WH one (a solution that tends to work without fail, by the way). However, I myself too do not at all understand why this problem was happening given the only difference between the `en` and `wh` scripts was whitespace, and whiespace *normally* does not affect this formatting tag in any way like this, and there are multiple `{a:c:` tags right in this very script that have whitespace in them.
@vit9696, could you clarify what you meant by "split it over segments"? The scripts were identical on these lines with the sole exception of a trailing space on both lines mentioned.

**Comment by vit9696** [comment]
```
`{a:c:text}`
`{a:c:continues}`
```

vs

```
`{a:c:text`
`continues}`
```

The latter is what I called split segments. Given the fix you pushed it may be a genuine bug in onscripter-ru, but I am not sure we will risk investigating as changes in the parser may break other things.

**Comment by ooa113y** [comment]
From what I understand, it needs to have one `{a:c:` per "real" line (as in, the line in the final produced `.txt` file), rather than one line in the localisation files.
The final line for this looks like this:
```
d `{a:c:`[lv 0*"27"*"60701044"]`......“Mother”?}`[\]
```
And this works fine.
Further down the script we have several `{a:c:` commands where it might look like one is enough, but that's because they are split over several "real" lines, rather than just in the localisation:
```
*d49494
d `{a:c:——Beatrice. `[@]`{n}Listen. `[@]`{n}I am giving you my love for Ushiromiya Battler. `[@]`{n}Become the kind of woman Battler wants.} `[\]
*d49495
d `{a:c:Take the golden hair he likes. `[@]`{n}Take the blue eyes he likes. `[@]`{n}Take this personality that matches his. `[@]`{n}And then, ...you must love him in my place. `[@]`{n}And then, if possible, you must be loved by him.} `[\]
*d49496
d `{a:c:I...can no longer love him. `[@]`{n}Please, take my feelings that went unfulfilled, `[@]`{n} ...my feelings that I couldn't hold back, `[!w583]`{n}...and make my wish come true.} `[\]
```
It's pretty clear to me that whatever was used to extract the localised scripts just grabs the different things between the opening and the closing backticks, and then inserts them as individual lines in the localised file.
---
So, tl;dr, I understand what you were saying and where the confusion came from. If this happens to be an engine bug with handling whitespace (presumably relevant: this is also an NVL-styled section of the script, so maybe that's why?) I understand if you'd rather not touch it, so I'll just close this since the actual problem is gone now.
